Bugsnag/BugsnagSystemState.m

```diff
@@ -243,6 +243,10 @@ - (instancetype)initWithConfiguration:(BugsnagConfiguration *)config {
 }
 
 - (void)sessionUpdateNotification:(NSNotification *)notification {
+    if (notification.object == nil) {
+        // No data to update
+        return;
+    }
     if (![BSGJSONSerialization isValidJSONObject:notification.object]) {
```

Comment:
Ah I think this is a genuine problem - when a session is stopped or paused, `BugsnagSystemState` should clear `state[BSGKeySession]` so that it no longer thinks there is an active session. Currently we would end up reporting a session when we shouldn't... this would affect OOM and thermal kill events.

```suggestion
    if (notification.object && ![BSGJSONSerialization isValidJSONObject:notification.object]) {
```

Reply:
Note: we have no test coverage for this edge case, calling `pauseSession` and causing an OOM / thermal kill.